platform/lib/routers/pages.js

```diff
@@ -156,6 +143,15 @@ if (config.isDevMode()) {
   });
 
   pages.get('/*', async (request, response, next) => {
+    // Grow has problems delivering the index.html on a root request
```

Comment:
can you extract this into a method as the exact same code is also used below

Reply:
Good idea: went into `ensureFileExtension`.